de.fu_berlin.inf.dpp.intellij/build.gradle

```diff
@@ -0,0 +1,101 @@
+buildscript {
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+  dependencies {
+    classpath "gradle.plugin.org.jetbrains.intellij.plugins:gradle-intellij-plugin:0.3.4"
+  }
+}
+
+apply plugin: 'org.jetbrains.intellij'
+
+dependencies {
+  compile project(':de.fu_berlin.inf.dpp.core')
+  compile project(':de.fu_berlin.inf.dpp.ui')
+  compile project(path: ':de.fu_berlin.inf.dpp.ui', configuration: 'testing')
+
+  compile 'org.easytesting:fest-assert:1.2'
+  compile 'org.easytesting:fest-reflect:1.2'
+  compile 'org.easytesting:fest-swing:1.2.1'
+  compile 'org.easytesting:fest-util:1.1.3'
+
+  testCompile project(path: ':de.fu_berlin.inf.dpp.core', configuration: 'testing')
+  testCompile configurations.testConfig
+}
+
+sourceSets {
+  main {
+    java.srcDir 'src'
+    resources.srcDir 'resources'
+
+    resources {
+      srcDir 'src'
+      exclude '**/*.java'
+    }
+  }
+
+  test.java.srcDir 'test/junit'
+}
+
+test {
+  systemProperty 'log4j.configurationFile', '../de.fu_berlin.inf.dpp.core/log4j2.xml'
+  exclude '**/Abstract*'
+}
+
+jar {
+  manifest {
+    instruction 'Created-By', 'IntelliJ IDEA'
+    instruction 'Manifest-Version', '1.0'
+  }
+  version = null
+  into ('META-INF')
+  {
+    from 'META-INF/plugin.xml'
+  }
+  from 'src/saros.log4j.properties'
+  from 'log4j2.properties'
+
+  from sourceSets.main.output
+}
+
+intellij {
+  // Download intellij only if the home property is not set
+  if (project.hasProperty('intellijHome')) {
```

Comment:
I am torn whether giving the developer a choice which IDE to use is necessary/useful. On the one hand, downloading a budeled version of IntelliJ wastes diskspace (about 1.2 gb per budle as the zip is not removed after extraction). On the other hand, using a set bundled version ensures that all developers are building against the same IDE version. Furthermore, I kind of like the workflow of having a pre-defined IDE version that can be updated and tested locally and then checked into git and pushed to the master, thereby making it clear which IDE version is supported and automatically updating which version the server builds with.

An important question regarding the additional diskspace needed to download budeled IDE versions is how gradle handles no longer needed dependencies. If I set a new IDE version in the config file, is the old version removed from the gradle cache? Otherwise, downloading IDE bundles would wast a *lot* of diskspace fast, depending on how closely we follow the IntelliJ release schedule (whether we also include patch releases).

Reply:
For now I would like to define the used intellij version outside of gradle. The version which is used is therefore define in the dockerfiles repo. Furthermore, using a different handling for eclipse and intellij would be very confusing.   

The old IDE versions are not removed.